server/statistics/buckets/bucket_stat_informer.go

```diff
@@ -0,0 +1,206 @@
+		b.RegionID, core.HexRegionKeyStr(b.StartKey), core.HexRegionKeyStr(b.EndKey), b.HotDegree, b.Interval, b.Loads)
+}
+
+// BucketTreeItem is the item of the bucket btree.
+type BucketTreeItem struct {
+	regionID uint64
+	startKey []byte
+	endKey   []byte
+	stats    []*BucketStat
+	interval uint64
+	version  uint64
+	status   status
+}
+
+// GetStartKey returns the start key of the bucket tree.
+func (b *BucketTreeItem) GetStartKey() []byte {
+	return b.startKey
+}
+
+// GetEndKey return the end key of the bucket tree item.
+func (b *BucketTreeItem) GetEndKey() []byte {
+	return b.endKey
+}
+
+// String implements the fmt.Stringer interface.
+func (b *BucketTreeItem) String() string {
+	return fmt.Sprintf("[region-id:%d][start-key:%s][end-key:%s]",
+		b.regionID, core.HexRegionKeyStr(b.startKey), core.HexRegionKeyStr(b.endKey))
+}
+
+// Less returns true if the start key is less than the other.
+func (b *BucketTreeItem) Less(than btree.Item) bool {
+	return bytes.Compare(b.startKey, than.(*BucketTreeItem).startKey) < 0
+}
+
+// compareKeyRange returns whether the key range is overlaps with the item.
+func (b *BucketTreeItem) compareKeyRange(origin *BucketTreeItem) bool {
+	if origin == nil {
+		return false
+	}
+	// key range must be same if the version is same.
+	if b.version == origin.version {
+		return true
+	}
+	return bytes.Equal(b.startKey, origin.startKey) && bytes.Equal(b.endKey, origin.endKey)
+}
+
+// cloneBucketItemByRange returns a new item with the same key range.
+// item must have some debris for the given key range
+func cloneBucketItemByRange(b *BucketTreeItem, startKey, endKey []byte) *BucketTreeItem {
+	item := &BucketTreeItem{
+		regionID: b.regionID,
+		startKey: startKey,
+		endKey:   endKey,
+		interval: b.interval,
+		version:  b.version,
+		stats:    make([]*BucketStat, 0, len(b.stats)),
+		status:   archive,
+	}
+
+	for _, stat := range b.stats {
+		//  insert if the stat has debris with the key range.
+		left := maxKey(stat.StartKey, startKey)
+		right := minKey(stat.EndKey, endKey)
+		if bytes.Compare(left, right) < 0 {
+			copy := stat.clone()
+			copy.StartKey = left
+			copy.EndKey = right
+			item.stats = append(item.stats, copy)
+		}
+	}
+	return item
+}
+
+// inherit the hot stats from the old item to the new item.
+// rule1: if one cross buckets are hot , it will inherit the hottest one.
+// rule2: if the cross buckets are not hot, it will inherit the coldest one.
+// rule3: if some cross buckets are hot and the others are cold, it will inherit the hottest one.
+func (b *BucketTreeItem) inherit(origins []*BucketTreeItem) {
+	if len(origins) == 0 || len(b.stats) == 0 || bytes.Compare(b.endKey, origins[0].startKey) < 0 {
+		return
+	}
+
+	newItems := b.stats
+	oldItems := make([]*BucketStat, 0)
+	for _, bucketTree := range origins {
+		oldItems = append(oldItems, bucketTree.stats...)
+	}
+	// details: https://leetcode.cn/problems/interval-list-intersections/solution/jiu-pa-ni-bu-dong-shuang-zhi-zhen-by-hyj8/
```

Comment:
Can we use some comments to describe this problem?

Reply:
fix